### PR TITLE
LIMS-6632: Orders: Contact configuration approach: In case the user configures the contact field to display name & number this action should reflect on the order active table

### PR DESCRIPTION
**Pull request by omnia353**
```
C:\Users\omnia\Documents\1limsNewMaster\1lims-automation>nosetests -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test006_orders.py -m test094 --tc-file=config.ini
c:\users\omnia\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlugin: No module named 'fcntl'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-09-16 12:19:56.732 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-16 12:19:56.737 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-16 12:19:57.782 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : WEMx9il-XjuLzKvHDm8it_SqSiyzUYssL5501_lFJAQ .....

DevTools listening on ws://127.0.0.1:52981/devtools/browser/36e11f4d-815f-4707-9e2b-30c1dcda8881
test094_change_contact_config_0_Name (ui_testing.testcases.basic_tests.test006_orders.OrdersTestCases) ...
2020-09-16 12:21:10.496 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test094_change_contact_config_0_Name
2020-09-16 12:21:21.346 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:21:21.440 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:21:22.592 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:21:23.759 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:427 - set test unit configuration
2020-09-16 12:21:24.804 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-16 12:21:24.812 | INFO     | api_testing.apis.orders_api:set_configuration:432 - set order configuration
2020-09-16 12:21:25.839 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-16 12:21:25.861 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3246 - get random contact
2020-09-16 12:21:25.872 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/contacts
2020-09-16 12:21:27.004 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-16 12:21:27.012 | INFO     | ui_testing.pages.orders_page:open_order_config:345 - open testunits configurations
2020-09-16 12:21:28.163 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3251 - change contact view by options
2020-09-16 12:21:28.167 | INFO     | ui_testing.pages.contacts_page:open_contact_configurations_options:137 - open contact configurations options
2020-09-16 12:21:28.185 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:21:30.206 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:21:32.037 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
['Name']
2020-09-16 12:21:33.118 | INFO     | ui_testing.pages.contacts_page:clear_all_selected_view_and_search_options:145 - clear selected view and search options of contact name
2020-09-16 12:21:33.333 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:21:42.760 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-16 12:21:45.233 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-16 12:21:48.043 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3267 - go to order active table
2020-09-16 12:21:51.347 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:21:52.878 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:21:53.839 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:21:55.291 | INFO     | ui_testing.pages.orders_page:filter_by_contact:163 - Filter by contact : OMNYA
2020-09-16 12:22:04.307 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:22:05.446 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:06.389 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:07.600 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-16 12:22:11.072 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok
test094_change_contact_config_1_No (ui_testing.testcases.basic_tests.test006_orders.OrdersTestCases) ...
2020-09-16 12:22:11.078 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test094_change_contact_config_1_No
2020-09-16 12:22:14.522 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:22:14.580 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:15.832 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:17.146 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:427 - set test unit configuration
2020-09-16 12:22:18.958 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-16 12:22:18.967 | INFO     | api_testing.apis.orders_api:set_configuration:432 - set order configuration
2020-09-16 12:22:21.008 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-16 12:22:21.015 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3246 - get random contact
2020-09-16 12:22:21.023 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/contacts
2020-09-16 12:22:21.417 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-16 12:22:21.425 | INFO     | ui_testing.pages.orders_page:open_order_config:345 - open testunits configurations
2020-09-16 12:22:22.463 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3251 - change contact view by options
2020-09-16 12:22:22.467 | INFO     | ui_testing.pages.contacts_page:open_contact_configurations_options:137 - open contact configurations options
2020-09-16 12:22:22.470 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:24.665 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:26.631 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
['No']
2020-09-16 12:22:27.623 | INFO     | ui_testing.pages.contacts_page:clear_all_selected_view_and_search_options:145 - clear selected view and search options of contact name
2020-09-16 12:22:27.703 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:35.889 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-16 12:22:37.489 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-16 12:22:39.935 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3267 - go to order active table
2020-09-16 12:22:42.933 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:22:42.953 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:43.741 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:44.730 | INFO     | ui_testing.pages.orders_page:filter_by_contact:163 - Filter by contact : 1366
2020-09-16 12:22:54.990 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:22:55.312 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:56.082 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:22:56.859 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-16 12:22:59.117 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok
test094_change_contact_config_2_Name_No (ui_testing.testcases.basic_tests.test006_orders.OrdersTestCases) ...
2020-09-16 12:22:59.120 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test094_change_contact_config_2_Name_No
2020-09-16 12:23:00.133 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:23:01.111 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:01.882 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:02.671 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:427 - set test unit configuration
2020-09-16 12:23:04.174 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-16 12:23:04.177 | INFO     | api_testing.apis.orders_api:set_configuration:432 - set order configuration
2020-09-16 12:23:05.111 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-16 12:23:05.117 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3246 - get random contact
2020-09-16 12:23:05.118 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/contacts
2020-09-16 12:23:05.280 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-16 12:23:05.288 | INFO     | ui_testing.pages.orders_page:open_order_config:345 - open testunits configurations
2020-09-16 12:23:05.815 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3251 - change contact view by options
2020-09-16 12:23:05.819 | INFO     | ui_testing.pages.contacts_page:open_contact_configurations_options:137 - open contact configurations options
2020-09-16 12:23:05.820 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:07.223 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:08.189 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
['Name', 'No']
2020-09-16 12:23:08.954 | INFO     | ui_testing.pages.contacts_page:clear_all_selected_view_and_search_options:145 - clear selected view and search options of contact name
2020-09-16 12:23:09.022 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:24.468 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-16 12:23:26.057 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-16 12:23:28.001 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test094_change_contact_config:3267 - go to order active table
2020-09-16 12:23:29.368 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:23:29.866 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:30.696 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:31.687 | INFO     | ui_testing.pages.orders_page:filter_by_contact:163 - Filter by contact : OMNYA
2020-09-16 12:23:42.034 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-16 12:23:42.499 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:43.266 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-16 12:23:44.175 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-16 12:23:45.304 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 3 tests in 228.524s

OK

```

### CODE REVIEW

File: ui_testing/elements.py [hunk 1]
```diff
@@ -558,6 +558,8 @@
                        'value': '//*[@id="custom-accordion-panel"]/div/a/i'},
         'archive': {'method': 'link_text',
                     'value': 'Archive'},
+        'configurations': {'method': 'link_text',
```
Comment (ghada-37) [inline ui_testing/elements.py:561]: it's already defined in general so no need to re-define it
https://github.com/modeso-open-source/1lims-automation/blob/ba7e0d580646ba7cceec19fc3da33c00f1fd1b0f/ui_testing/elements.py#L111

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 2]
```diff
@@ -3265,6 +3266,46 @@ def test094_select_large_number_of_test_units_in_one_testplan(self):
...
+        elif search_by == 'Name:No':
+            search_text = payload['name'] + ' No: ' + str(payload['companyNo'])
+            search_by = search_by.split(':')
+            result = payload['name']
+
+        self.contacts_page.select_option_to_view_search_with(view_search_options=search_by)
+        self.info('go to order active table')
+        self.orders_page.get_orders_page()
+        self.orders_page.filter_by_contact(filter_text=result)
+        rows = self.orders_page.result_table()
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:3305]: here you need to make sure that value assigned to cell contact exactly equal (name, no, name+no) 
so you need to use :
https://github.com/modeso-open-source/1lims-automation/blob/4b510411856316f9b3e5ae4715a4d2a9a1d7558a/ui_testing/pages/base_selenium.py#L671

**Review by ghada-37 — CHANGES_REQUESTED** [review]

File: ui_testing/pages/contacts_page.py [hunk 3]
```diff
@@ -133,3 +133,27 @@ def search_find_row_open_edit_page(self, search_text):
         rows = self.search(search_text)
         self.open_edit_page_by_css_selector(row=rows[0])
 
+    def open_contact_configurations_options(self):
+        self.info('open contact configurations options')
+        self.sleep_tiny()
```
Comment (0xIslamTaha) [inline ui_testing/pages/contacts_page.py:138]: remove it

File: ui_testing/pages/contacts_page.py [hunk 4]
```diff
@@ -133,3 +133,27 @@ def search_find_row_open_edit_page(self, search_text):
...
+        self.base_selenium.click(element='configurations_page:field_options')
+        self.sleep_tiny()
+
+    def clear_all_selected_view_and_search_options(self):
+        self.info('clear selected view and search options of contact name')
+        self.base_selenium.click(element='configurations_page:clear_all')
+        self.sleep_tiny()
+
+    def select_option_to_view_search_with(self, view_search_options):
+        print(view_search_options)
```
Comment (0xIslamTaha) [inline ui_testing/pages/contacts_page.py:150]: replace print with `info`

**Review by 0xIslamTaha — CHANGES_REQUESTED** [review]

**Review by 0xIslamTaha — APPROVED** [review]

**Pull request merged**